src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/model/managedindexmetadata/ActionMetaData.kt

```diff
@@ -76,6 +76,17 @@ data class ActionMetaData(
         return Strings.toString(this, false, false)
     }
 
+    fun asTemplateArg(): Map<String, Any?> {
```

Comment:
Same comment. Lets try to reuse `toXContent` instead of creating new function which can have different logic. I already see that this is missing `ActionProperties`.